src/dbnode/persist/fs/retriever.go

```diff
@@ -341,8 +343,10 @@ func (r *blockRetriever) fetchBatch(
 			seg = ts.NewSegment(data, nil, checksum, ts.FinalizeHead)
 		}
 
+		blockCachingEnabled := r.opts.CacheBlocksOnRetrieve() && r.nsCacheBlocksOnRetrieve
```

Comment:
so you have to set both the global and namespace option? what is the point of the global option if you have to enable the namespace?

Reply:
Probably makes more sense to think about it the other way around. Both of these options default to true opting you in to caching (which is how it works today). But to disable caching, you can either turn it off globally with one knob or disable for a specific namespace with a single knob.

Reply:
nit: can pull `r.opts.CacheBlocksOnRetrieve()` into a var outside of the loop too; in fact can compute `blockCachingEnabled` outside